azurerm/internal/services/compute/tests/resource_arm_dedicated_host_test.go

```diff
@@ -0,0 +1,239 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/compute/parse"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMDedicatedHost_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dedicated_host", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDedicatedHostDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDedicatedHost_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "license_type", string(compute.DedicatedHostLicenseTypesNone)),
+					resource.TestCheckResourceAttr(data.ResourceName, "auto_replace_on_failure", "true"),
+					resource.TestCheckResourceAttr(data.ResourceName, "platform_fault_domain", "1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "sku_name", "DSv3-Type1"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDedicatedHost_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dedicated_host", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDedicatedHostDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDedicatedHost_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "platform_fault_domain", "1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "sku_name", "DSv3-Type1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "license_type", string(compute.DedicatedHostLicenseTypesWindowsServerHybrid)),
+					resource.TestCheckResourceAttr(data.ResourceName, "auto_replace_on_failure", "false"),
```

Comment:
instead of checking the value of the specific fields we can just use the import step below to do this for us